src/frontend/src/elements/YearSwitcher.tsx

```diff
@@ -0,0 +1,39 @@
+import { useSearchParams } from "react-router-dom";
+
+
+function getCurrentYear() {
+    return new Date().getFullYear();
+}
+
+const LAST = -1;
+
+
+function useYearManipulation(): [number, (y: number) => void] {
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    function setYear(year: number) {
+        setSearchParams((prev) => {
+            prev.set("year", `${year}`);
+            return prev;
+        });
+    }
+
+    const years = searchParams.getAll("year");
+    const year = years.length ? parseInt(years.at(LAST)!) : getCurrentYear();
```

Comment:
```suggestion
    const year = years.length ? parseInt(years.at(LAST)) : getCurrentYear();
```
typo? oder macht das was?

Reply:
typescript denkt noch, das man `null` zurückbekommen könnte (was nicht möglich ist durch den check davor). das `!` macht, dass `undefined` und `null` aus der liste der möglichen datentypen entfernt wird